microsoft-edge/progressive-web-apps-chromium/how-to/index.md

```diff
@@ -1,113 +1,139 @@
+You can find the final source code of the app you will be building in this guide on the [PWA getting started demo app repository](https://github.com/MicrosoftEdge/Demos/tree/main/pwa-getting-started).
 
 
 <!-- ====================================================================== -->
 ## Prerequisites
 
 *   Install [Visual Studio Code](https://code.visualstudio.com) to edit your PWA source code.
 *   Install [Node.js](https://nodejs.org) to use it as your local web server.
+*   Working knowledge of HTML, CSS, and JavaScript is also a plus.
 
 
 <!-- ====================================================================== -->
-## Creating a basic website
+## The architecture of a PWA
 
-In this step, you will create a simple HTML page to create a basic website, and will start a local web server to access the website in your browser.
+Progressive Web Apps are written using the programming languages of the web: HTML, CSS, and JavaScript, and are distributed to your users by using web servers.
 
-1. Create a new directory called `MySamplePWA` on your computer.
+To make your app available to users, you deploy it on a web server that's accessible via HTTPS. Your server contains:
 
-1. Open Visual Studio Code, select **File** > **Open Folder** and then select the `MySamplePWA` directory to open your new project.
+* **Back-end code**: the endpoints needed by your app, when connected to the internet, to retrieve dynamic content that may be stored in a database on your server.
+* **Front-end code**: the resources needed for the app to be installed on the user's device, such as HTML, CSS, and JavaScript code.
 
-1. Create a new file in the project by pressing `Ctrl+N`, add the following content, and save the file as `index.html`:
+Your back-end code can use the server-side languages of your choice such as ASP.NET, Java, Node.js, or PHP. Note, however, that server-side endpoints may not even be required depending on the app your are building. The PWA you create in this tutorial does not have any server-side code because it exclusively runs on the device it's installed on and doesn't need any server-side data.
 
-   ```html
-   <!DOCTYPE html>
-   <html lang="en-US" dir="ltr">
-     <head>
-       <meta charset="UTF-8" />
-       <meta name="viewport" content="width=device-width,initial-scale=1" />
-       <link rel="shortcut icon" href="https://c.s-microsoft.com/favicon.ico?v2" />
-       <title>My Sample PWA</title>
-     </head>
-     <body>
-       <h1>My Sample PWA</h1>
-     </body>
-   </html>
-   ```
+Your front-end code uses HTML, CSS, JavaScript, and JSON only.
 
-1. Start a local web server by using the `http-server` Node.js library:
+You use HTML to describe the content in your app, such as the text, images, text fields, or buttons that appear in the user interface. You then use CSS to organize the HTML content in a layout, and provide styles to elements. And finally you use JavaScript to add user interactions to your user interface.
```

Comment:
mention json here, because it's mentioned above.  What's the role of JSON files or JSON-formatted messages?

Reply:
Great point. I'll add the JSON manifest thing at the end of this paragraph. It's important to keep here because the architecture diagram further down has it too.